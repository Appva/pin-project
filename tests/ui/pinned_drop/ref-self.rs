// `ref (mut) self` are rejected by rustc.

use std::pin::Pin;

pub struct Struct {
    field: u8,
}

impl Struct {
    fn take_ref_self(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
    fn take_ref_mut_self(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
}

fn main() {}
