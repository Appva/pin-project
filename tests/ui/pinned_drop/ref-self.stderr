error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:10:26
   |
10 |     fn take_ref_self(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                          ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
  --> $DIR/ref-self.rs:11:34
   |
11 |     fn take_ref_mut_self(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
   |                                  ^^^^ expected identifier, found keyword
